infra/DockerRepoManager.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class to manage a git repos interaction with a docker image.
+
+This class is to be used to manage Oss-Fuzz projects docker build images.
+It allows for a build image to be build with a specific commit rather
+than just the current head of the repo.
+
+  Typical usage example:
+
+    drm = DockerRepoManager('curl')
+    current_commit = drm.get_image_commit()
+    drm.set_image_commit('df26f5f9c36e19cd503c0e462e9f72ad37b84c82')
+
+"""
+import os
+
+from helper import _build_image
+from helper import _check_project_exists
+from helper import _get_dockerfile_path
+from helper import _is_base_image
+from RepoManager import RepoManager
+
+
+class NoRepoFoundError(Exception):
+  """Occurs when the bisector cant infer the main repo."""
+  pass
+
+
+class DockerRepoManagerError(Exception):
+  """When there is a docker error of execution"""
+  pass
+
+
+class DockerRepoManager(RepoManager):
```

Comment:
Since this is inheriting from RepoManager, we should implement this class in such a way that we can use the interface as defined by RepoManager, without having to implement additional logic. 

i.e. override checkout_commit like so:

```python
def checkout_commit(self, commit):
  super(DockerRepoManager, self).checkout_commit(commit)
  self.set_image_commit()   #
```

That said, we we simply change to mounting rather than copying in srcs every time (as discussed in another comment), we can probably get rid of this class completely and make things much simpler.